### PR TITLE
DuckDB Extension

### PR DESCRIPTION
**Pull request by eddyxu**
Initialize support of Duckdb Extension

### CODE REVIEW

File: integration/duckdb/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Lance DuckDB Extension
+
+*Linux Only*
```
Comment (changhiskhan) [inline integration/duckdb/README.md:3]: add for now?

File: integration/duckdb/README.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+## Functions
+
+List / Array functions
+
+| Function            | Description                        |
+|---------------------|------------------------------------|
+| `list_argmax(list)` | Run `argmax` over a list of values |
```
Comment (changhiskhan) [inline integration/duckdb/README.md:11]: add this is only for floats? Let's also file a duckdb github issue. This should just be a native duckdb function

File: integration/duckdb/README.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
+# Lance DuckDB Extension
```
Comment (changhiskhan) [inline integration/duckdb/README.md:1]: need to add a summary description

File: integration/duckdb/src/lance/duckdb/vector_functions.cc [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+                               L2Distance<float>));
+  l2_distance.AddFunction(
+      ::duckdb::ScalarFunction({::duckdb::LogicalType::LIST(::duckdb::LogicalType::DOUBLE),
+                                ::duckdb::LogicalType::LIST(::duckdb::LogicalType::DOUBLE)},
+                               ::duckdb::LogicalType::DOUBLE,
+                               L2Distance<double>));
+  functions.emplace_back(std::make_unique<::duckdb::CreateScalarFunctionInfo>(l2_distance));
+
+  ::duckdb::ScalarFunctionSet rect_contains("rect_contains");
+  rect_contains.AddFunction(::duckdb::ScalarFunction(
```
Comment (changhiskhan) [inline integration/duckdb/src/lance/duckdb/vector_functions.cc:89]: let's also support int/long rectangle regions? i.e., `SELECT rect_contains(tsne, [[0, 0], [300, 300]] FROM dataset`

Comment (changhiskhan) [reply integration/duckdb/src/lance/duckdb/vector_functions.cc:89]: and let's either:

1. change the order of the arguments (usually `rect_contains(a, b)` is interpreted as "the rectangle a contains b", which is the reverse

OR

2. change the name of the function to `in_rectangle`, so it's `a is in rectangle b`

Comment (eddyxu) [reply integration/duckdb/src/lance/duckdb/vector_functions.cc:89]: ok, changed it to `in_rectangle` for clarity.

File: integration/duckdb/README.md (hunk 2 above)
Comment (eddyxu) [reply integration/duckdb/README.md:11]: It supports `int/int64/float/double`.

Sure can file an issue upstream.

**Review by changhiskhan — APPROVED** [review]
assume this still works locally and that unit tests are forthcoming

File: integration/duckdb/README.md (hunk 1 above)
Comment (eddyxu) [reply integration/duckdb/README.md]: done

**Pull request merged**